executable-spec/src/Cardano/Ledger/Spec/STS/Update/Tallysip.hs

```diff
@@ -196,11 +196,13 @@ tallyOutcome vres stakeDistribution pNoQ pNoM r_a =
                  &&
                  Data.stakePercentRound (Data.stakeAbstain vres) (totalStake stakeDistribution)
                  <= vThreshold r_a
-                 && Data.rvNoMajority vres <= pNoM
+                 && Data.rvNoMajority vres < pNoM
                 then
                   Data.NoMajority
                 else
-                  Data.Expired
+                  if Data.rvNoQuorum vres >= pNoQ || Data.rvNoMajority vres >= pNoM
+                    then Data.Expired
+                    else error "Tallysip.tallyOutcome: Reached a problemativ tally outcome."
```

Comment:
We need to provide more information about why a "problematic" tally outcome was found. Imagine if you're faced with this error: you'd like to know more :)

Reply:
Reached an unpredictable tally outcome.
It is an outcome that should not appear ever. We have to see if it ever comes up in our tests

Reply:
But still, if it comes we would be left with an unhelpful error message. If possible we should give more information about:

- why is problematic
- what is the context (e.g. voting counters)